pkg/utils/transform/transform.go

```diff
@@ -109,6 +113,98 @@ func InjectLabel(key, value string, overwritePolicy OverwritePolicy, kinds ...st
 	}
 }
 
+func DeploymentImages(images map[string]string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "Deployment" {
+			return nil
+		}
+
+		d := &appsv1.Deployment{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, d)
+		if err != nil {
+			return err
+		}
+
+		containers := d.Spec.Template.Spec.Containers
+		replaceContainerImages(containers, images)
+
+		unstrObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(d)
+		if err != nil {
+			return err
+		}
+		u.SetUnstructuredContent(unstrObj)
+
+		return nil
+	}
+}
+
+func replaceContainerImages(containers []corev1.Container, images map[string]string) {
+	for i, container := range containers {
+		name := strings.ToLower(container.Name)
+		if url, exist := images[name]; exist {
+			containers[i].Image = url
+		}
+
+		replaceContainersArgsImage(&container, images)
+	}
+}
+
+func replaceContainersArgsImage(container *corev1.Container, images map[string]string) {
+	for a, arg := range container.Args {
+		argument := strings.TrimPrefix(arg, "-")
+		argument = ARG_PREFIX + strings.ToLower(argument)
+		if url, exist := images[argument]; exist {
+			container.Args[a+1] = url
```

Comment:
just thinking out loud,
what if i write the arg list as 
`-arg=value` instead of `-arg value`

Reply:
Wise thought I'm hearing it. I will add the patch for it. I felt we could also file an issue on pipeline, triggers to accept it as `ENV VAR` or other way for accepting the configuration.